File: extensions/aliexpress/src/test/java/app/morphe/extension/aliexpress/patches/DisableForcedUpdatesPatchTest.java
package app.morphe.extension.aliexpress.patches;

import org.junit.Test;
import static org.junit.Assert.*;

public class DisableForcedUpdatesPatchTest {

    @Test
    public void shouldShowUpdateDialog_alwaysReturnsFalse() {
        assertFalse(DisableForcedUpdatesPatch.shouldShowUpdateDialog());
    }

    @Test
    public void getMinSupportedVersionCode_returnsMaxInt() {
        assertEquals(Integer.MAX_VALUE, DisableForcedUpdatesPatch.getMinSupportedVersionCode(1));
        assertEquals(Integer.MAX_VALUE, DisableForcedUpdatesPatch.getMinSupportedVersionCode(999999));
        assertEquals(Integer.MAX_VALUE, DisableForcedUpdatesPatch.getMinSupportedVersionCode(-1));
    }
}

File: patches/src/test/kotlin/app/morphe/patches/aliexpress/updates/DisableForcedUpdatesPatchTest.kt
package app.morphe.patches.aliexpress.updates

import org.junit.Assert.assertEquals
import org.junit.Assert.assertNotNull
import org.junit.Assert.assertTrue
import org.junit.Test

class DisableForcedUpdatesPatchTest {

    @Test
    fun verifyUpdateCheckFingerprintBoundaries() {
        val fingerprint = UpdateCheckFingerprint

        assertEquals("V", fingerprint.returnType)
        assertNotNull(fingerprint.strings)
        assertEquals(1, fingerprint.strings?.size)
        assertTrue(fingerprint.strings?.contains("checkUpdate") ?: false)
    }

    @Test
    fun verifyDisableForcedUpdatesPatchProperties() {
        val patch = disableForcedUpdatesPatch

        assertEquals("Disable forced updates", patch.name)
        assertEquals("Disables the forced update popup that prevents using older versions of the app.", patch.description)
        assertTrue(patch.default)
    }
}
